InvenTree/users/models.py

```diff
@@ -1 +1,314 @@
 # -*- coding: utf-8 -*-
+
+from django.contrib.auth.models import Group, Permission
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+
+
+class RuleSet(models.Model):
+    """
+    A RuleSet is somewhat like a superset of the django permission  class,
+    in that in encapsulates a bunch of permissions.
+
+    There are *many* apps models used within InvenTree,
+    so it makes sense to group them into "roles".
+
+    These roles translate (roughly) to the menu options available.
+
+    Each role controls permissions for a number of database tables,
+    which are then handled using the normal django permissions approach.
+    """
+
+    RULESET_CHOICES = [
+        ('admin', _('Admin')),
+        ('part', _('Parts')),
+        ('stock', _('Stock')),
+        ('build', _('Build Orders')),
+        ('purchase_order', _('Purchase Orders')),
+        ('sales_order', _('Sales Orders')),
+    ]
+
+    RULESET_NAMES = [
+        choice[0] for choice in RULESET_CHOICES
+    ]
+
+    RULESET_MODELS = {
+        'admin': [
+            'auth_group',
+            'auth_user',
+            'auth_permission',
+            'authtoken_token',
+            'users_ruleset',
+        ],
+        'part': [
+            'part_part',
+            'part_bomitem',
+            'part_partcategory',
+            'part_partattachment',
+            'part_partsellpricebreak',
+            'part_parttesttemplate',
+            'part_partparametertemplate',
+            'part_partparameter',
+        ],
+        'stock': [
+            'stock_stockitem',
+            'stock_stocklocation',
+            'stock_stockitemattachment',
+            'stock_stockitemtracking',
+            'stock_stockitemtestresult',
+        ],
+        'build': [
+            'part_part',
+            'part_partcategory',
+            'part_bomitem',
+            'build_build',
+            'build_builditem',
+            'stock_stockitem',
+            'stock_stocklocation',
+        ],
+        'purchase_order': [
+            'company_company',
+            'part_supplierpart',
```

Comment:
`SupplierPart` and `SupplierPartBreak` are models of the `Company` app, not the `Part` app.

Reply:
They were *originally* in the Part app, so the database tables retain the old names:

![image](https://user-images.githubusercontent.com/10080325/95027384-c9ec9500-06e3-11eb-8cf5-bad9969a836a.png)

Reply:
And the unit testing has been setup to detect any of the following cases

- A ruleset references a permission that does not exist
- An existing permission is not referenced by any ruleset

Reply:
@SchrodingersGat I was not able to create a new `Group` after pulling your branch, I hit the `ContentType.DoesNotExist` error...

Reply:
You're right, it does not actually use the table name, it *does* use the app name.

So `company_supplierpart` and `company_supplierpricebreak` are correct here.

I'm just writing some better unit testing to check for this. I'll push it momentarily